### PR TITLE
improve test descriptions to be custimizable per problem

### PR DESCRIPTION
**Pull request by jitwit**

### CODE REVIEW

File: code/exercises/perfect-numbers/test.ss [hunk 1]
```diff
@@ -29,5 +29,6 @@
    `((test . ,(spec->tests spec))
      (version . ,(lookup 'version spec))
      (skeleton . "perfect-numbers.scm")
-     (solution . "example.scm"))))
+     (solution . "example.scm")
+     (hints.md . ,(splice-exercism 'perfect-numbers.scm)))))
```
Comment (jitwit) [inline code/exercises/perfect-numbers/test.ss]: todo, fix to `'perfect-numbers`

File: code/exercises/hamming/test.ss [hunk 2]
```diff
@@ -34,12 +34,12 @@
...
-        `((sentence "For scheme, you may want to look into one of "
-                    (inline-code "error")
-                    ", "
-                    (inline-code "assert")
-                    ", or "
-                    (inline-code "raise")
-                    ".")))))))
+      ,(splice-exercism 'hamming
+                        '(sentence "For scheme, you may want to look into one of "
+                                   (inline-code "error")
```
Comment (jitwit) [inline code/exercises/hamming/test.ss]: 1/2 of the actual changes

File: code/md.ss [hunk 3]
```diff
@@ -3,38 +3,69 @@
...
+          "Running and testing your solutions"
+          (subsection
+           "Overview"
+           (enum
+            (item "Start a REPL either in your favorite editor or from
+the command line.")
+            (item "Type "
+                  (inline-code ,(format "(load \"~a.scm\")" exercism))
+                  " at the prompt.")
+            (item "Test your code by calling "
```
Comment (jitwit) [inline code/md.ss]: other 1/2 of actual changes

File: code/md.ss [hunk 4]
```diff
@@ -86,4 +117,8 @@
 (define (md-hints md)
   `(section "Notes" ,@md))
 
+(define (splice-exercism problem . md)
+  `((section "Track Specific Notes" ,@md)
+    (exercism-test-help ,problem)))
+
```
Comment (jitwit) [inline code/md.ss]: other half of actual changes

File: config/exercise_readme.go.tmpl [hunk 5]
```diff
@@ -4,9 +4,6 @@
 {{- with .Hints }}
 {{ . }}
 {{ end }}
-{{- with .TrackInsert }}
-{{ . }}
-{{ end }}
```
Comment (guygastineau) [inline config/exercise_readme.go.tmpl]: If this is now getting spliced in house, then perhaps the code for doing that could take a list of files to read and splice in.  In that case we wouldn't need to copy the `TESTS.md` file at all, and we could just reference it directly.

Comment (guygastineau) [reply config/exercise_readme.go.tmpl]: Oh nevermind, I see that you elected to include altered notes on how to run the tests.

File: .gitignore [hunk 6]
```diff
@@ -6,3 +6,4 @@ bin/configlet.exe
 *~
 _build/*
 *.html
+ci
```
Comment (guygastineau) [inline .gitignore]: cool

**Comment by guygastineau** [comment]
Seems fine

File: config/exercise_readme.go.tmpl (hunk 5 above)
Comment (jitwit) [reply config/exercise_readme.go.tmpl]: Yeah, I think that could be a good next step. I think we should still transition incrementally, though

File: .gitignore (hunk 6 above)
Comment (jitwit) [reply .gitignore]: hah yeah, that was bugging me

**Pull request merged**